### ISSUE TITLE
Monthly reports

### ISSUE THREAD

**Issue by robs74** [robs74/offices-generate-money#1]
There should be monthly reports about money earned in each currency.

**Comment by marcelinorc** [comment]
This is a great point! I can help. Any ideas on how to implement this?

**Comment by robs74** [comment]
I've seen some monthly reports in scripts by Chałwa, maybe it would be possible to adapt them to our case.

**Comment by robs74** [comment]
Yeah, it may turn out that this is actually impossible at this point because variables cannot be transferred between different script runs. We may need to postpone it. The devs promised to add such variables later. For now, I added a daily balance instead as an objective, the script now runs once a day instead of fifteen times a day to make the income/loss easier to plan.

**Comment by marcelinorc** [comment]
Interesting. If you point me in the direction of Chalwa scripts I can  dig it out. Making the script run less often will mean that you only account for the amount of workers you have once a day so in an extreme case in the very moment you run the script you have no workers you would make no money, even if you did have workers on the office during the day.

**Comment by marcelinorc** [comment]
Interesting. If you point me in the direction of Chalwa scripts I can  dig it out. Making the script run less often will mean that you only account for the amount of workers you have once a day so in an extreme case in the very moment you run the script you have no workers you would make no money, even if you did have workers on the office during the day.

**Comment by marcelinorc** [comment]
Interesting. If you point me in the direction of Chalwa scripts I can  dig it out. Making the script run less often will mean that you only account for the amount of workers you have once a day so in an extreme case in the very moment you run the script you have no workers you would make no money, even if you did have workers on the office during the day.

**Comment by robs74** [comment]
No no, I was mistaken. I think you can't make it at this point, or at least it'd have to be very convoluted. This is also the main reason why I changed the tick. Not much we can do at the moment I'm afraid. When they add such "global" variables we can get back to more frequent calculations as well as introduce monthly reports.

**Comment by robs74** [comment]
No no, I was mistaken. I think you can't make it at this point, or at least it'd have to be very convoluted. This is also the main reason why I changed the tick. Not much we can do at the moment I'm afraid. When they add such "global" variables we can get back to more frequent calculations as well as introduce monthly reports.

**Comment by marcelinorc** [comment]
I think it is possible to compute the amount of money made by the office by subtracting the stats from the amount of money the player has.

**Comment by robs74** [comment]
Hmm, stats don't store your treasury, only income and costs. Using stats requires assuming that they register all your balance changes except offices, so this mod immediately becomes incompatible with others which change money. Moreover, I think you can't get the cost of inviting workers (only the number of them). That would be also really ugly as you'd have to sum over all the resources manually 😅 If these imperfections don't scare you, you can try.

I already use objectives in the new version, but they only show the amount per tick. I don't think I can update them let's say every 15 ticks with the sum or take a trailing approach with storing the 15 last ticks and summing them.
![20230327121336_1](https://user-images.githubusercontent.com/31097629/228133521-0b5f9bb3-e143-488a-bbfe-56c27baa41d5.jpg)

**Comment by chalwachalwa** [comment]
@marcelinorc :
This one: 
https://github.com/chalwachalwa/Rasputin-challenge-pyrois-script
https://github.com/chalwachalwa/Aerial-delivery---pyrois-script

Only solution for now would be while loop and storing that data in script. Checking date everyday (I used 50 seconds not 60 to make sure that check will occure at least once a day, with 60 that could be tricky and bug in some corner cases I guess, no possibilty to test that).

**Comment by chalwachalwa** [comment]
There is a function to set how many instructions are executed (Script_SetUpdateFrequency) per frame? per tick? I don't know. There is also critical section instruction to explicitly tell game to execute some isntructions in one frame. If it works that would be a solution, performance over timing. This is not Real Time System we can't exactly specify how long execution will take

**Comment by robs74** [comment]
When using the infinite loop approach it's possible to store variables, but I still couldn't create a notification because strings are cumbersome here.
